Automattic/jetpack
# Search dashboard refactoring

#### Changes proposed in this Pull Request:
Refactored the components to be more intuitive and moved pages to a folder to be more clear. The PR also ensures only necessary API call are made. For example, only pricing API is called on connection page and  upsell page.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no
#### Testing instructions:
- Create a new JN site with current branch ([Special OP with JP Beta plugin](https://jurassic.ninja/specialops/))
- Ensure connection page, upsell page, dashboard page (including record meter) all work well
